packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -1964,6 +1964,10 @@ function commitMutationEffects_begin(
     if (deletions !== null) {
       for (let i = 0; i < deletions.length; i++) {
         const childToDelete = deletions[i];
+        // When deleting a host node, we follow the return path to find the
+        // neartest mounted host parent. Which means the return pointer of
+        // the deleted tree must be correct.
+        childToDelete.return = fiber;
```

Comment:
In my testing, the `return` pointer was _not_ null the first time something was deleted. It ended up being null later b'c we tried to delete the same thing twice.

I'm wondering if the repro I had (the larger one that involved Feed and was more touch and go) is a different problem.

Reply:
It’s always non-null at this particular location, but before the fix it would sometimes point to the wrong alternate. Does that describe what you were seeing?

Reply:
Hm, pretty sure I was seeing a double deletion, and the pointer was cleared in detach mutation.

Reply:
You're right, I checked again and it was a double deletion.

Fix here: https://github.com/facebook/react/pull/20942